tests/utils.py

```diff
@@ -137,6 +138,36 @@ def generate_core_file(
             process.wait(timeout=TIMEOUT)
 
 
+@contextlib.contextmanager
+def generate_core_at_breakpoint(
+    python: pathlib.Path,
+    test_file: pathlib.Path,
+    tmpdir: pathlib.Path,
+) -> Generator[pathlib.Path, None, None]:
+    gdb_commands = f"""
+set pagination off
+run
+generate-core-file {tmpdir}/core.%p
+quit
+"""
```

Comment:
nit, but I don't like seeing stuff flushed to the left column. Can we do:
```suggestion
    gdb_commands = f"""
        set pagination off
        run
        generate-core-file {tmpdir}/core.%p
        quit
    """
```
Or, if gdb isn't ok with leading whitespace, can we add a textwrap.dedent?